src/dataset-builder/DatasetBuilderUtils.ts

```diff
@@ -85,6 +87,7 @@ export type DatasetAccessRequestApi = {
 /** Below are the UI types */
 
 export interface DomainCriteria extends DomainCriteriaApi, Criteria {
+  conceptId: number;
```

Comment:
What is the difference between `CriteriaApi.id` and `DomainCriteria.conceptId`? Do we need both?

Reply:
`CriteriaApi.id` defines the domainId or ID for the ProgramDataOption.